### DISCUSSION TITLE
I want to use get_tables, which contains this method

### DISCUSSION THREAD

**Discussion by juzj** [pymupdf/PyMuPDF#2312] [Looking for help]
My requirement is to extract the text in pdf. If it is the deletion or skip of the table, save the result as txt

But the latest version doesn't have this feature, so I can use that method instead

**Comment by JorjMcKie** [comment]
I do not fully understand:
Do you mean to add a feature that automatically extracts a table from the text of a document page?

**Comment by juzj** [comment]
I want to delete all the table data from the pdf

**Comment by juzj** [comment]
But now without the get_tables method, I can't tell if the current data is a table

**Comment by juzj** [comment]
<img width="1584" alt="image" src="https://user-images.githubusercontent.com/36066382/228800140-61873688-fdbf-463c-8d70-bb015c830570.png">

I will cycle through all the pages, if there is tabular data to skip without processing, what do I need to do?

**Comment by JorjMcKie** [comment]
I think I understand now.
Your request implies the need for a capability to **_understand_** what a table **_is_** in the first place.
This is a highly complex task  - for which there exists no general solution - nowhere in the software market. There are however some approaches that come close to doing this, which employ advanced Artificial Intelligence and Machine Learning features.
While PyMuPDF's extraction features deliver a large set of information on a document page's contents, it will remain your task to programatically interpret these data.
So you must use your knowledge about a given page to determine a table's whereabouts (like certain keywords in its header text or any footer lines, etc.) and then exclude the respective text from your extraction.

**Comment by JorjMcKie** [comment]
Let me first move your post to the "Discussions" tab - where it belongs.

**Comment by JorjMcKie** [comment]
You seem to be lucky:
The tables on your example page are all wrapped by rectangle drawings. PyMuPDF has a way to extract vector graphics - use `Page.get_drawings()`. Please do read the [documentation](https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_drawings).
The method returns a list of dictionaries, which each contain atomic draw commands (like draw a line, draw a curve, etc.), plus a rectangle that encapsulates those draw commands.
So I suggest you extract the drawings of your page and confirm that you will see the 4 table rectangles being represented by a drawing dictionary each. Do a few experimenting here.
Once confirmed, check whether your extracted text is contained in one of these rectangles and if so skip it.

**Reply by juzj** [reply]
Thank you. I'll try.

**Reply by JorjMcKie** [reply]
Please do not hesitate to ask if you need help.

**Comment by Nicolas2912** [comment]
I have encountered a related problem. I'm using `get_drawings()` to extract vector graphics from a PDF page, but the function also returns tables, which I want to ignore. Is there a recommended way to distinguish between actual drawings and tables in the output? Any suggestions on filtering or post-processing the extracted data would be appreciated!

**Reply by JorjMcKie** [reply]
You could use `page.cluster_drawings()`. This works on top of `.get_drawings()` trying to find interconnected / neighboring / overlapping vector graphics and returns a list of rectangles that wrap more or less isolated vector graphics.
This could save you a lot programming effort.

Once you have this rectangle list, execute `.find_tables()` and check each `table.bbox`. If contained in one of the above clusters, you have what you are looking for.
